pennylane/plugins/default_qubit.py

```diff
@@ -379,11 +362,7 @@ def expval(self, observable, wires, par):
             ev = self.ev(A, wires)
         else:
             # estimate the ev
-            # sample Bernoulli distribution n_eval times / binomial distribution once
-            a, P = spectral_decomposition_qubit(A)
-            p0 = self.ev(P[0], wires)  # probability of measuring a[0]
-            n0 = np.random.binomial(self.shots, p0)
-            ev = (n0*a[0] +(self.shots-n0)*a[1]) / self.shots
+            ev = np.mean(self.sample(observable, wires, par, self.shots))
```

Comment:
What do you think about making this the default implementation for the expval function in the Device base class?

Reply:
As in, return this line in `Device.expval`? We could even do the same for `Device.var`, and require plugins to only support `Device.sample`.

However, my preference would be to continue keeping `Device.expval` and `Device.var` as an abstract method, as certain frameworks have built in approaches for computing expectation values (c.f. `operator_estimation` in PyQuil).

Reply:
Good point, I'll also have a look at the var implementation of default.qubit.

Actually, if I came in here at this point to develop a new plugin I would not understand why I'm forced to implement var and expval by hand if it's just `self.sample(...).var()` and `self.sample(...).mean()`. I totally see why people would want to override it, but the default can just be the simple behaviour, can't it?

Reply:
The main exception I can think of is for devices that are wavefunction simulators (i.e., `default.qubit`, `forest.wavefunction`, Qiskit `StatevectorSimulator`). In this case, you would want to bypass the sampling to return exact results for the default `shots=0`. In this case, I like that it is explicit to make it a required method to be implemented.

In the case of hardware simulators that must interface via sampling, I agree that it makes sense to have `sample().mean()` be the default.

Reply:
Good! I mean people can just have a look at `default.qubit` and see how this is done.

Reply:
However, I feel this will require some more discussion - let's make an issue for this for more discussion, and implement potential changes in a new PR.

We have been thinking of having plugin devices 'register' themselves with PennyLane as a state simulator (i.e., wavefunction access) or hardware simulator/hardware device, in which case we might be able to choose sensible defaults for both cases.